SCons/Node/NodeTests.py

```diff
@@ -711,30 +711,30 @@ def test_set_always_build(self) -> None:
         node = SCons.Node.Node()
         node.set_always_build()
         assert node.always_build
-        node.set_always_build(3)
-        assert node.always_build == 3
```

Comment:
These things are weird, in the code `noclean`, `nocache`, `precious` and `always_build` are used like bools, but there are some comments about leaving a couple of them as ints, and the unit tests seem to treat them as ints that could have values other than 0/1 (`None` seems to be possible too in some cases).  That's probably not necessary because of the way that code is written - the `IDX` function the tree printer uses always converts to a `bool` anyway.

Reply:
The comments appear to come from a time before the `IDX` function was a thing, and no other parts of the code appear to utilize these for anything other than boolean evaluation. The only exception was this test file, but it doesn't seem to actually be to any end; making them evaluate truthiness instead felt more applicable

Reply:
> The comments appear to come from a time before the `IDX` function was a thing, and no other parts of the code appear to utilize these for anything other than boolean evaluation. The only exception was this test file, but it doesn't seem to actually be to any end; making them evaluate truthiness instead felt more applicable

perhaps I'll take a whack at converting them in a future PR.